### PR TITLE
Adding extra include tests and enabling some more that were switched off.

### PR DESCRIPTION
**Pull request by maumar**

### CODE REVIEW

**Comment by maumar** [comment]
@anpete

File: test/Microsoft.EntityFrameworkCore.FunctionalTests/ComplexNavigationsQueryTestBase.cs [hunk 1]
```diff
@@ -229,7 +229,7 @@ public virtual void Multi_level_include_with_short_circuiting()
         }
 
         // issue #3186
```
Comment (maumar) [inline test/Microsoft.EntityFrameworkCore.FunctionalTests/ComplexNavigationsQueryTestBase.cs]: will remove this comment

File: test/Microsoft.EntityFrameworkCore.FunctionalTests/ComplexNavigationsQueryTestBase.cs [hunk 2]
```diff
@@ -296,7 +296,7 @@ public virtual void Join_navigation_key_access_required()
...
```
Comment (maumar) [inline test/Microsoft.EntityFrameworkCore.FunctionalTests/ComplexNavigationsQueryTestBase.cs]: and this one

**Comment by anpete** [comment]
:shipit:

**Pull request closed without merging**